File: 101-250/169-majority-element.cpp
int majorityElement(vector<int>& nums) {
    int n = nums.size();
    int m = nums[0], count = 0;
    for (int i = 0; i < n; ++i) {
        if (nums[i] == m) {
            ++count;
        }
        else if (count == 0) {
            m = nums[i];
            ++count;
        }
        else {
            --count;
        }
    }

    return m;
}

File: python version/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 1
        k = nums[0]
        for i in range(1, len(nums)):
            if k == nums[i]:
                count += 1
            else:
                count -= 1
                if count == 0:
                    k = nums[i]
                    count = 1
        return k